backend/src/main/java/com/benine/backend/PresetController.java

```diff
@@ -13,6 +14,8 @@
 
   private ArrayList<Preset> presets = new ArrayList<Preset>();
 
+  private HashSet<String> tags = new HashSet<>();
```

Comment:
We also have a static keywords set in the preset class.
I think we have to choose one of them not both.